web/js/doppelkopf.js

```diff
@@ -5,19 +5,42 @@ function addPlayers(select, players) {
     option.appendChild(document.createTextNode(player.name));
     select.appendChild(option);
   }
+  for (let i = 1; i < 5; i += 1) { // Vorauswahl auf empty setzen
+    document.getElementById(`player${i}`).selectedIndex = -1;
+  }
 }
 
 window.onload = function onLoad() {
   const request = new XMLHttpRequest();
 
   request.open('GET', '/api/players');
   request.addEventListener('load', function onRequestLoad() {
-    if (request.status >= 200 && request.status < 300) {
-      const players = JSON.parse(request.responseText);
-      addPlayers(document.getElementById('player'), players);
-    } else {
-      console.warn(request.statusText, request.responseText);
+    for (let i = 1; i < 5; i += 1) { // 4 Spieler können ausgewählt werden
+      if (request.status >= 200 && request.status < 300) {
+        const players = JSON.parse(request.responseText);
+        addPlayers(document.getElementById(`player${i}`), players);
+      } else {
+        console.warn(request.statusText, request.responseText);
+      }
     }
   });
   request.send();
 };
+
+document.addEventListener('DOMContentLoaded', function storeChoice() { // Auswahl des Spielers wird gespeichert
+  for (let i = 1; i < 5; i += 1) {
+    const input = document.getElementById(`player${i}`);
+    if (localStorage[`player${i}`]) {
+      input.value = localStorage[`player${i}`];
+    }
+    input.onchange = function storeChange() {
+      localStorage[`player${i}`] = this.value;
```

Comment:
Üblicherweise würdest du das `target` des Events (also das `<select>` Feld, das sich geändert hat) nicht über `this` sondern mit `event.target` abgreifen. Also
```
input.onchange = function storeChange(event) {
  localStorage[`player${i}`] = event.target.value;
}
```
Das Problem ist, das `this` nicht immer verlässlich das `<select>` Element zurückgibt.